internal/config/validate.go

```diff
@@ -12,20 +12,23 @@ func (c *Config) validate() error {
 		errs.appendFieldError("BUILDKITE_SPLITTER_RETRY_COUNT", "was %d, must be greater than or equal to 0", c.MaxRetries)
 	}
 
-	if got, min := c.Parallelism, 1; got < min {
+	if c.Parallelism == nil {
+		errs.appendFieldError("BUILDKITE_PARALLEL_JOB_COUNT", "not set in environment")
+	} else if got, min := *c.Parallelism, 1; got < min {
```

Comment:
Should we keep the max validation?
```suggestion
	if c.Parallelism == nil {
		errs.appendFieldError("BUILDKITE_PARALLEL_JOB_COUNT", "not set in environment")
	} else {
	  if got, min := *c.Parallelism, 1; got < min {
	    ...
	  }
	  if got, max := *c.Parallelism, 1000; got > max {
	    ...
	  }
	}
```

Reply:
ohh good catch! will add the max validation back